modules/costs/app/contracts/time_entries/base_contract.rb

```diff
@@ -121,11 +124,11 @@ def validate_activity_active
     end
 
     def work_package_invisible?
-      model.work_package.nil? || !model.work_package.visible?(user)
+      model.entity.nil? || !model.entity.visible?(user)
```

Comment:
`model.entity.nil?` will always be false.

Reply:
Since we're delegating `entity`:

```suggestion
      !entity.visible?(user)
```

(this method could probably be inlined)

Reply:
Why do you assume `entity.nil?` will always be false? You can call the contract with a time entry that has no entity set.

Reply:
See above: https://github.com/opf/openproject/pull/18665/files#diff-0f591a25204ca2455d89cbacf9fbe4fdb37a56bcf490809f125277e1addeef63R95